### PR TITLE
API endpoints to change a keep's users + libraries

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@cvializ I'll walk you through the format

### CODE REVIEW

File: kifi-backend/modules/shoebox/conf/admin.routes [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+##########################################
+# Using a get so we can invoke from browser and have it take care of the auth part
+GET     /admin/server               @com.keepit.controllers.admin.AdminHealthController.serviceView()
+GET     /admin/benchmark            @com.keepit.controllers.admin.AdminBenchmarkController.benchmarks()
+GET     /admin/healthcheck/error    @com.keepit.controllers.admin.AdminHealthController.causeError()
+GET     /admin/healthcheck/reportDeployment    @com.keepit.controllers.admin.AdminHealthController.reportDeployment()
+GET     /admin/healthcheck/causeAirbrakeError @com.keepit.controllers.admin.AdminHealthController.causeAirbrakeError(message: String ?= "[No Worry] Fake Airbrake Exception triggered from admin")
+GET     /admin/healthcheck/emailErrors         @com.keepit.controllers.admin.AdminHealthController.reportErrors()
+GET     /admin/healthcheck/errors              @com.keepit.controllers.admin.AdminHealthController.getErrors()
+GET     /admin/healthcheck/resetErrorCount     @com.keepit.controllers.admin.AdminHealthController.resetErrorCount()
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/conf/admin.routes:323]: I bought us another month or so :pray:

**Comment by ryanpbrewster** [comment]
@andrewconner @leogrim @camhashemi I'm merging this in 20 minutes (I have actually negative interest in dealing with merge conflicts due to splitting out admin routes). Speak now or forever hold your peace.

**Comment by andrewconner** [comment]
Andrew likes it.

**Comment by leogrim** [comment]
I feel like we'd be better off with separate APIs for adding and removing members

**Comment by ryanpbrewster** [comment]
Consider what would happen if you want to change the library a keep is in. One way is to have 3 endpoints, `add`, `remove`, and `move`. Or we could have one, `modify`. IMO it is cleaner to have one.

**Comment by leogrim** [comment]
Actually I'm good with it.

**Comment by leogrim** [comment]
Now I insist that `KeepConnections` is just too generic of a name. `KeepMembers` is not perfect but closer to the product, and we agreed on that with Derek.

File: kifi-backend/modules/common/app/com/keepit/model/KeepConnections.scala [hunk 2]
```diff
@@ -59,3 +63,17 @@ object KeepMembers {
...
+object KeepConnectionsDiff {
+  def addUser(user: Id[User]) = KeepConnectionsDiff(users = DeltaSet.empty.add(user), libraries = DeltaSet.empty)
+  def addUsers(users: Set[Id[User]]) = KeepConnectionsDiff(users = DeltaSet.empty.addAll(users), libraries = DeltaSet.empty)
+  def addLibrary(library: Id[Library]) = KeepConnectionsDiff(users = DeltaSet.empty, libraries = DeltaSet.empty.add(library))
+}
+case class ExternalKeepConnectionsDiff(users: DeltaSet[ExternalId[User]], libraries: DeltaSet[PublicId[Library]])
+object ExternalKeepConnectionsDiff {
+  implicit val reads: Reads[ExternalKeepConnectionsDiff] = (
+    (__ \ 'users).readNullable[DeltaSet[ExternalId[User]]].map(_ getOrElse DeltaSet.empty) and
+    (__ \ 'libraries).readNullable[DeltaSet[PublicId[Library]]].map(_ getOrElse DeltaSet.empty)
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/model/KeepConnections.scala:77]: `emails` ?

Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/model/KeepConnections.scala:77]: soon! ((I hope))

File: kifi-backend/modules/shoebox/conf/admin.routes (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/conf/admin.routes:323]: :raised_hands:

**Pull request merged**